var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -223,106 +230,94 @@ def libs(self):
-            else:
-                extra_args.append('--disable-fortran2003')
+        if '+static' in spec:
+            args.append('-DONLY_SHARED_LIBS=OFF')
+        else:
+            args.append('-DONLY_SHARED_LIBS=ON')
+
+        if '+debug' in spec:
+            args.append('-DCMAKE_BUILD_TYPE=Debug')
+        else:
+            args.append('-DCMAKE_BUILD_TYPE=Release')
+
+        if '+threadsafe' in spec:
+            args.append('-DHDF5_ENABLE_THREADSAFE=ON')
+        else:
+            args.append('-DHDF5_ENABLE_THREADSAFE=OFF')
+
+        if '+hl' in spec:
+            args.append('-DHDF5_BUILD_HL_LIB=ON')
+        else:
+            args.append('-DHDF5_BUILD_HL_LIB=OFF')
+
+        if '+cxx' in spec:
+            args.append('-DHDF5_BUILD_CPP_LIB=ON')
+        else:
+            args.append('-DHDF5_BUILD_CPP_LIB=OFF')
 
-        if '+shared' in self.spec:
-            extra_args.append('--enable-shared')
+        if '+fortran' in spec:
+            args.append('-DHDF5_BUILD_FORTRAN=ON')
         else:
-            extra_args.append('--disable-shared')
-            extra_args.append('--enable-static-exec')
-
-        if '+pic' in self.spec:
-            extra_args.extend([
-                'CFLAGS='   + self.compiler.cc_pic_flag,
-                'CXXFLAGS=' + self.compiler.cxx_pic_flag,
-                'FCFLAGS='  + self.compiler.fc_pic_flag,
-            ])
-
-        if '+mpi' in self.spec:
-            # The HDF5 configure script warns if cxx and mpi are enabled
-            # together. There doesn't seem to be a real reason for this, except
-            # that parts of the MPI interface are not accessible via the C++
-            # interface. Since they are still accessible via the C interface,
-            # this is not actually a problem.
-            extra_args += ['--enable-parallel',
-                           'CC=%s' % self.spec['mpi'].mpicc]
-
-            if '+cxx' in self.spec:
-                extra_args.append('CXX=%s' % self.spec['mpi'].mpicxx)
-
-            if '+fortran' in self.spec:
-                extra_args.append('FC=%s' % self.spec['mpi'].mpifc)
-
-        extra_args.append('--with-zlib=%s' % self.spec['zlib'].prefix)
-
-        return extra_args
-
-    @run_after('configure')
-    def patch_postdeps(self):
-        if '@:1.8.14' in self.spec:
-            # On Ubuntu14, HDF5 1.8.12 (and maybe other versions)
-            # mysteriously end up with "-l -l" in the postdeps in the
-            # libtool script.  Patch this by removing the spurious -l's.
-            filter_file(
-                r'postdeps="([^"]*)"',
-                lambda m: 'postdeps="%s"' % ' '.join(
-                    arg for arg in m.group(1).split(' ') if arg != '-l'),
-                'libtool')
+            args.append('-DHDF5_BUILD_FORTRAN=OFF')
+
+        if '+java' in spec:
+            args.append('-DHDF5_BUILD_JAVA=ON')
+        else:
+            args.append('-DHDF5_BUILD_JAVA=OFF')
+
+        if '+tools' in spec:
```

Comment:
I guess you forgot to declare variant `tools`. Do you think we need it?

Reply:
On line 65.  Not building tools reduces build time and binary size.